### ISSUE TITLE
CAN drivers not building/working for green_wp750x

### ISSUE THREAD

**Issue by magiconair** [mangOH/mangOH#10]
I am trying to build CAN drivers for a `green_wp750x`. The mcp251x can bus card is in slot 1 since we have the WiFi module in slot 0.

The most recent `mangOH.sdef` includes the drivers but only for the `red_xxxx` (https://github.com/mangOH/mangOH/blob/master/mangOH.sdef#L128-L152).

When I add the drivers to the green without any further changes (i.e. not commenting out `spisvc`) it builds 

```
mangoh@mangoh-vm:~/mangOH$ make clean green_wp750x
mangoh@mangoh-vm:~/mangOH$ find build -name *.ko
build/green_wp750x/staging/modules/can_iot.ko
build/green_wp750x/staging/modules/can-raw.ko
build/green_wp750x/staging/modules/lsm6ds3-i2c.ko
build/green_wp750x/staging/modules/mangoh_green_dv4.ko
build/green_wp750x/staging/modules/can-dev.ko
build/green_wp750x/staging/modules/can.ko
build/green_wp750x/staging/modules/lsm6ds3.ko
build/green_wp750x/staging/modules/spisvc.ko
build/green_wp750x/staging/modules/vcan.ko
build/green_wp750x/staging/modules/can-bcm.ko
build/green_wp750x/staging/modules/mcp251x.ko
build/green_wp750x/modules/spisvc/spisvc.ko
build/green_wp750x/modules/lsm6ds3/lsm6ds3.ko
build/green_wp750x/modules/can/can.ko
build/green_wp750x/modules/mangoh_green_dv4/mangoh_green_dv4.ko
build/green_wp750x/modules/vcan/vcan.ko
build/green_wp750x/modules/can_iot/can_iot.ko
build/green_wp750x/modules/lsm6ds3-i2c/lsm6ds3-i2c.ko
build/green_wp750x/modules/mcp251x/mcp251x.ko
build/green_wp750x/modules/can-dev/can-dev.ko
build/green_wp750x/modules/can-bcm/can-bcm.ko
build/green_wp750x/modules/can-raw/can-raw.ko
```

... but the subsequent installation fails:

```
Jul 17 09:46:27 swi-mdm9x15 user.err Legato: =ERR= | gpioExpanderService[2157]/gpioExpanderCommon T=main | gpioExpander.c I2cAccessBusAddr() 972 | Could not open file /dev/i2c-6 or /dev/i2c/6: No such file or directory
Jul 17 09:46:27 swi-mdm9x15 user.err Legato: =ERR= | gpioExpanderService[2157]/gpioExpanderCommon T=main | gpioExpander.c SmbusWriteReg() 1053 | failed to open i2c bus 6 for access to address 63
Jul 17 09:46:27 swi-mdm9x15 user.emerg Legato: *EMR* | gpioExpanderService[2157]/gpioExpanderCommon T=main | gpioExpander.c gpioExpander_Reset() 812 | Failed to reset GPIO expander on I2C bus 6 at address 0x3f
Jul 17 09:46:28 swi-mdm9x15 user.warn Legato: -WRN- | atClientDaemon[1953]/framework T=main | LE_FILENAME le_fdMonitor_Create() 672 | FD Monitor object name 'atClientDaemon.watchdogChain.le_wdog' truncated to 'atClientDaemon.watchdogChain.le'.
Jul 17 09:46:29 swi-mdm9x15 user.warn Legato: -WRN- | atServerDaemon[1954]/framework T=main | LE_FILENAME le_fdMonitor_Create() 672 | FD Monitor object name 'atServerDaemon.watchdogChain.le_wdog' truncated to 'atServerDaemon.watchdogChain.le'.
Jul 17 09:46:29 swi-mdm9x15 user.info Legato:  INFO | gpioService[2012]/sysfsGpio T=main | gpioSysfsUtils.c gpioSysfs_SessionCloseHandlerFunc() 1178 | Releasing GPIO 2
Jul 17 09:46:29 swi-mdm9x15 user.info Legato:  INFO | supervisor[1905]/supervisor T=main | proc.c proc_SigChildHandler() 2035 | Process 'gpioExpanderService' (PID: 2157) has exited with exit code 1.
Jul 17 09:46:29 swi-mdm9x15 user.emerg Legato: *EMR* | supervisor[1905]/supervisor T=main | proc.c proc_SigChildHandler() 2077 | Process 'gpioExpanderService' reached fault limit while system in probation. Device will be rebooted.
Jul 17 09:46:29 swi-mdm9x15 user.warn Legato: -WRN- | _appStopClient[2192]/framework T=main | LE_FILENAME CreateSocket() 550 | Socket opened as standard i/o file descriptor 2!
Jul 17 09:46:29 swi-mdm9x15 user.info Legato:  INFO | modemDaemon[2024]/swiQmi T=main | swiQmi.c InitService() 378 | Init QMI service QMI_SERVICE_DMS (2)
Jul 17 09:46:29 swi-mdm9x15 user.info Legato:  INFO | modemDaemon[2024]/swiQmi T=main | swiQmi.c PrintServiceDetails() 354 | Service.2: library_version.6 idl_version.1 service_id.0x2 idl_minor_version.35
Jul 17 09:46:30 swi-mdm9x15 user.info Legato:  INFO | modemDaemon[2024]/swiQmi T=main | swiQmi.c StartService() 113 | qmi_client_get_service_list rc=0, numServices=1
Jul 17 09:46:30 swi-mdm9x15 user.info Legato:  INFO | modemDaemon[2024]/swiQmi T=main | swiQmi.c InitService() 378 | Init QMI service QMI_SERVICE_UIM (8)
Jul 17 09:46:30 swi-mdm9x15 user.info Legato:  INFO | modemDaemon[2024]/swiQmi T=main | swiQmi.c PrintServiceDetails() 354 | Service.8: library_version.6 idl_version.1 service_id.0xB idl_minor_version.42
Jul 17 09:46:30 swi-mdm9x15 user.emerg Legato: *EMR* | supervisor[1905]/supervisor T=main | app.c app_SigChildHandler() 3494 | Process 'gpioExpanderService' in app 'gpioExpanderServiceGreen' faulted: Rebooting system.
Jul 17 09:46:30 swi-mdm9x15 user.emerg Legato: *EMR* | supervisor[1905]/supervisor T=main | supervisor.c framework_Reboot() 695 | Supervisor going down to trigger reboot.

```

When I comment out `$LEGATO_ROOT/drivers/spisvc/spisvc` as described in the comment the build fails.

```
make[2]: Leaving directory '/home/mangoh/legato_framework/legato'
make[1]: Leaving directory '/home/mangoh/legato_framework/legato'
TOOLCHAIN_DIR=/opt/swi/y17-ext-wp750x/sysroots/x86_64-pokysdk-linux/usr/bin/arm-poky-linux-gnueabi \
TOOLCHAIN_PREFIX=arm-poky-linux-gnueabi- \
MANGOH_BOARD=GREEN \
mksys -t wp750x --object-dir=build/green_wp750x --output-dir=build/update_files/green mangOH.sdef
** Warning: Ignoring empty app specification
** Warning: Ignoring empty app specification
** ERROR:
Kernel module spisvc.mdef must be listed in sdef file.
Makefile:47: recipe for target 'green_wp750x' failed
make: *** [green_wp750x] Error 1
```

I am using Legato`18.05.1_a37432c091f4345ee50613db47b0da5c` and the latest mangOH code (https://github.com/mangOH/mangOH/commit/f58376a72bc907a1cf2967669918790d04529e7e) and reproduce this in the mangOH VM. 

What am I missing?

**Comment by dpfrey** [comment]
This should be fixed in 39aba910a70468629d5efab8203a5e8bb8a73f76.  Please let me know if you're still stuck.

**Comment by magiconair** [comment]
We have given up on using mangOH and Legato since the toolchain is too complex, the Sierra Wireless dependencies too heavy and outdated, and the internal RPC mechanism not friendly towards different languages. I'm coming from Go where cross-compilation is a matter of setting an environment variable.

We chose a vendor which allows us to use our toolchain while providing a good hardware platform abstraction and integrated brick-free software updates.
